src/aro/Driver.zig

```diff
@@ -892,3 +899,225 @@ fn exitWithCleanup(d: *Driver, code: u8) noreturn {
     }
     std.process.exit(code);
 }
+
+/// Parses the various -fpic/-fPIC/-fpie/-fPIE arguments.
+/// Then, smooshes them together with platform defaults, to decide whether
+/// this compile should be using PIC mode or not.
+/// Returns a tuple of ( backend.CodeGenOptions.PicLevel, IsPIE).
+pub fn getPICMode(d: *Driver, args: []const []const u8) !struct { backend.CodeGenOptions.PicLevel, bool } {
```

Comment:
Is there a reason why the PIC related args can't be parsed in `parseArgs` before being handed to this function like other args?

Reply:
`getPICMode` involves the following parameters：
- `-fPIC/-fpic/-fpie/-fPIE/-fno-pic/-fno-PIC/-fno-pie/-fno-PIE,  -->  confirm which parameter last appeared in the compilation arguments，be parsed in the `parseArgs` function is ok。

(The following parameters arocc currently does not support it, it will be supported in the future, so a complete implementation has been provided based on the implementation in LLVM ). It's mainly these parameters that led me to directly use `args` in `getPICMode`

- `-mkernel`, `-fapple-kext`, `-mno-abicalls`  --> check if certain parameters exist in 'args'.
- `-fropi/-fno-ropi/-frwpi/-fno-rwpi`，`-mdynamic-no-pic`  --> confirm which parameter last appeared in the compilation arguments. 
- `-mcmodel=` --> required obtain the value after the equal.

Reply:
I think what @Vexu is saying (and I agree) is that it's probably better to structure it so that we only have to make one pass over the args array. So this would involve bringing back the if statements and driver fields that I said to take out in my last review (sorry), and add new fields for `kernel`, `apple_kext`, `ropi`, `rwpi`, etc. If you're not entirely sure how to handle it let me know and I'll push a commit to this branch.